keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/titus/verification/ContainerTestVerificationEvaluator.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.titus.verification
+
+import com.netflix.spinnaker.keel.api.Verification
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.plugins.VerificationEvaluator
+import com.netflix.spinnaker.keel.api.verification.VerificationContext
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.FAILED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.PASSED
+import com.netflix.spinnaker.keel.api.verification.VerificationStatus.RUNNING
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.titus.batch.ContainerJobConfig
+import com.netflix.spinnaker.keel.titus.batch.createRunJobStage
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.springframework.stereotype.Component
+
+@Component
+class ContainerTestVerificationEvaluator(
+  private val orca: OrcaService,
+  private val taskLauncher: TaskLauncher
+) : VerificationEvaluator<ContainerTestVerification> {
+
+  override val supportedVerification: Pair<String, Class<ContainerTestVerification>> =
+    "container-tests" to ContainerTestVerification::class.java
+
+  override fun evaluate(
+    context: VerificationContext,
+    verification: Verification,
+    metadata: Map<String, Any?>
+  ): VerificationStatus {
+    val taskId = metadata["taskId"]
+    require(taskId is String) {
+      "No task id found in previous verification state"
+    }
+
+    return runBlocking {
+      withContext(IO) {
+        orca.getOrchestrationExecution(taskId)
+      }
+        .let { response ->
+          when {
+            response.status.isSuccess() -> PASSED
+            response.status.isIncomplete() -> RUNNING
+            else -> FAILED
+          }
+        }
+    }
+  }
```

Comment:
Call to orca to see if the task is complete yet. We'll need to extract some context from the task response here if it fails.

Reply:
Just a suggestion to see what can be common between this implementation and `OrcaTaskMonitoringAgent`, which might be worth extracting into shared functions.

Reply:
Oh yeah, I forgot about that. I'll take a look.